conda/core/prefix_data.py

```diff
@@ -609,15 +632,17 @@ def set_environment_env_vars(
         self._write_environment_state_file(env_state_file)
         return env_state_file.get("env_vars")
 
-    def unset_environment_env_vars(
-        self, env_vars: dict[str, str]
-    ) -> dict[str, str] | None:
+    def unset_environment_env_vars(self, env_vars: list[str]) -> dict[str, str] | None:
+        self._ensure_no_reserved_env_vars(env_vars)
```

Comment:
Should this warning be displayed even when the user is unsetting the reserved env var? I think, maybe yes?

Reply:
IMO it does make sense to display the warning during unsetting as well for consistency.

Reply:
Reconsidering this, it doesn't really make sense to raise the warning when the user is trying to unset these reserved env vars. That action is actually fixing the configuration.